Core/Operator/Instance.cs

```diff
@@ -74,28 +74,28 @@ where typeof(IInputSlot).IsAssignableFrom(field.FieldType)
 
         public bool AddConnection(Symbol.Connection connection, int multiInputIndex)
         {
-            var (_, sourceSlot, _, targetSlot) = GetInstancesForConnection(connection);
-            if (targetSlot != null)
-            {
-                targetSlot.AddConnection(sourceSlot, multiInputIndex);
-                return true;
-            }
+            var foundInstance = TryGetInstancesForConnection(connection, out _, out var sourceSlot, out _, out var targetSlot);
 
-            return false;
+            if (!foundInstance)
+                return false;
+            
+            targetSlot.AddConnection(sourceSlot, multiInputIndex);
+            return true;
         }
 
         public void RemoveConnection(Symbol.Connection connection, int index)
         {
-            var (_, _, _, targetSlot) = GetInstancesForConnection(connection);
+            var foundInstance = TryGetInstancesForConnection(connection, out _, out _, out _, out var targetSlot);
+            if (!foundInstance)
+                return;
             targetSlot.RemoveConnection(index);
         }
 
-        private (Instance, ISlot, Instance, ISlot) GetInstancesForConnection(Symbol.Connection connection)
+        private bool TryGetInstancesForConnection(Symbol.Connection connection, out Instance sourceInstance, out ISlot sourceSlot, out Instance targetInstance, out ISlot targetSlot)
         {
             Instance compositionInstance = this;
 
-            var sourceInstance = compositionInstance.Children.SingleOrDefault(child => child.SymbolChildId == connection.SourceParentOrChildId);
-            ISlot sourceSlot;
+            sourceInstance = compositionInstance.Children.SingleOrDefault(child => child.SymbolChildId == connection.SourceParentOrChildId);
```

Comment:
opportunity for optimization - using `FirstOrDefault`. I did this with the `Symbol` and `SymbolUi` loading stuff, but thought these runtime checks might be necessary so I left them in. The only time I can see this ever needing to check for "Single" is at compile time - like if a dev were to make an operator from code and forget to make a guid unique

Reply:
Ah, so you would prefer `FirstOfDefault`? Good question...
I believe this would be premature. A better and much more substantial optimization would be to turn some of theses lists (esp. Symbol.Children) into ordered dictionaries.

Reply:
I'd definitely be a bigger fan of having these be ordered dictionaries too